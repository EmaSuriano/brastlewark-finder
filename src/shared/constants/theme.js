import PropTypes from 'prop-types';

export const ThemePropTypes = PropTypes.shape({
  background: PropTypes.string,
  light: PropTypes.string,
  disabled: PropTypes.string,
  accent: PropTypes.string,
  secondary: PropTypes.string,
}).isRequired;

export default {
  background: '#4F5D75',
  light: '#FFFFFF',
  disabled: '#BFC0C0',
  accent: '#EF8354',
  secondary: '#2D3142',
};
