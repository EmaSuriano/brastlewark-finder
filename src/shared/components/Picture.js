import styled from 'styled-components';
import PropTypes from 'prop-types';
import { ThemePropTypes } from 'shared/constants/theme';

const Picture = styled.img.attrs({
  atl: 'Gnome profile picture',
})`
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  border-radius: 50%;
  margin: auto;
  flex-shrink: 0;
  object-fit: cover;
  border: 1px ${props => props.theme.secondary} solid;
`;

Picture.propTypes = {
  theme: ThemePropTypes,
  size: PropTypes.number.isRequired,
};

export default Picture;
