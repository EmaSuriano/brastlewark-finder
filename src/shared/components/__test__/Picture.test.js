import React from 'react';
import { shallow } from 'enzyme';
import theme from 'shared/constants/theme';
import Picture from '../Picture';

describe('<Picture />', () => {
  describe('render', () => {
    it('with size', () => {
      expect(shallow(<Picture theme={theme} size={200} />)).toMatchSnapshot();
    });
  });
});
