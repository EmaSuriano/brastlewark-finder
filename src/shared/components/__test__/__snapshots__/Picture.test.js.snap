// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Picture /> render with size 1`] = `
.c0 {
  width: 200px;
  height: 200px;
  border-radius: 50%;
  margin: auto;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  object-fit: cover;
  border: 1px #2D3142 solid;
}

<img
  atl="Gnome profile picture"
  className="c0"
  size={200}
/>
`;
